073-decouple-ca-management-from-end-entity-certificate-management.md

```diff
@@ -0,0 +1,150 @@
+data:
+  ${fingerprint}.${state}: <PEM encoded CA certificate>
+```
+
+The `fingerprint` is the fingerprint/thumbprint of the certificate. I.e. the SHA-1 hash of the ASN.1 DER-encoded form of the X509 certificate, which is a commonly used way of identifying certificates using common tooling.
+The `state` can be any of the states of the CA certificate trust state machine described below.
+Including the state in the item name facilitates understanding the current role of the Kubernetes Secret within the cluster.
+
+The existing Kubernetes Secrets, named using `${cluster}-${ca}-certs` and passed via Secret volume mounts to containers would not change under this proposal.
+
+Problems from multiple processes concurrently updating the Kubernetes Secret are avoided by using `metadata.resourceVersion` to make conditional requests when updating that Secret. Note also that only one operator makes each state transition.
+
+### CA certificate state machine
+
+The "CA certificate trust state machine" has the following states:
+
+```
+  UNTRUSTED // not yet trusted everywhere it needs to be 
+   |
+   v
+  TRUSTED_UNUSED // trusted everywhere it needs to be, but no EE certificates yet issued with this CA certificate
+   |
+   v
+  TRUSTED_IN_USE // trusted everywhere, EE certificates issued using this CA certificate
+   |
+   v
+  PHASE_OUT // trusted everywhere, but new end-entity certificates will not be issued using this CA certificate
+   |
+   v
+  ZOMBIE // still trusted by some component, but with no issued end-entity certificates in current use.
+```
+
+New cluster creation process:
+1. The CaReconciler creates the CA certificates and places them in the relevant Kubernetes Secret with the state UNTRUSTED
+2. The cluster and user operator, during their respective reconcile loops:
+   1. Add the CA certificate to the relevant component truststore Kubernetes Secret and change the state to TRUSTED_UNUSED
+   2. Generate end entity certificates using the CA and add the EE certificate to the relevant component keystore Kubernetes Secret and change the state to TRUSTED_IN_USE
+
+Update process (e.g. CA private key has been updated):
+1. The CaReconciler generates a new CA certificate and places it in the relevant Kubernetes Secret with the state UNTRUSTED. It also updates the old CA certificate to have the state PHASE_OUT.
+2. The cluster and user operator, during their respective reconcile loop:
+   1. Add the CA certificate to the relevant component truststore Kubernetes Secret and change the state to TRUSTED_UNUSED
+   2. Generate end entity certificates using the CA and add the EE certificate to the relevant component keystore Kubernetes Secret and change the state to TRUSTED_IN_USE
+   3. Update the old CA certificate to ZOMBIE, indicating that no EE certicates are using the CA certificate
+3. In a future loop the CaReconciler removes the CA certificate from the shared Kubernetes Secret (because it is observed to be in the `ZOMBIE` state).
```

Comment:
```suggestion
3. In a future loop the CaReconciler removes the old CA certificate from the shared Kubernetes Secret (because it is observed to be in the `ZOMBIE` state).
```